File: Backend/Felino.Api/Domain/Entities/Product.cs
namespace Felino.Api.Domain.Entities
{
    public class Product
    {
        public int Id { get; set; }

        public string Name { get; set; } = "";

        public string Slug { get; set; } = "";

        public string Ingredients { get; set; } = "";

        public decimal Price { get; set; }

        public string? Sauce { get; set; }
        
        public string? AltText { get; set; }
        
        public string? ImageUrl { get; set; }

        public int CategoryId { get; set; }

        public Category? Category { get; set; }
    }
}

File: Frontend/client/src/types/product.ts
export type ProductDto = {
  id: number;
  name: string;
  slug: string;
  ingredients: string;
  price: number;
  sauce?: string | null;
  altText?: string | null;
  imageUrl?: string | null;
  categoryId?: number | null;
};

export type CreateProductDto = {
  name: string;
  ingredients: string;
  price: number;
  sauce?: string | null;
  altText?: string | null;
  imageUrl?: string | null;
  categoryId: number;
};

export type UpdateProductDto = {
  name?: string;
  ingredients?: string;
  price?: number;
  sauce?: string | null;
  altText?: string | null;
  imageUrl?: string | null;
  categoryId?: number | null;
};

export type FeaturedProductDto = {
  id: number;
  name: string;
  slug: string;
  ingredients: string;
  price: number;
  sauce?: string | null;
  altText?: string | null;
  imageUrl?: string | null;
  categoryId?: number | null;
  totalSold: number;
  isFallback: boolean;
};

export type ProductModalData = {
  id: number;
  name: string;
  image: string;
  category: string;
  ingredients?: string[];
  sides?: string[];
  sauce?: string;
  price: number;
};